internal/action/run_trigger.go

```diff
@@ -0,0 +1,156 @@
+package action
+
+import (
+	"context"
+	"fmt"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/takescoop/terraform-cloud-workspace-action/internal/tfconfig"
+	"github.com/takescoop/terraform-cloud-workspace-action/internal/tfeprovider"
+)
+
+type RunTriggerInput struct {
+	SourceID   string `yaml:"id"`
+	SourceName string `yaml:"name"`
+}
+
+type RunTriggerInputs []RunTriggerInput
+
+type RunTrigger struct {
+	SourceID  string
+	Workspace *Workspace
+	DataRef   map[string]tfeprovider.DataWorkspace
+}
+
+type RunTriggers []RunTrigger
+
+func (rt *RunTriggerInput) ToRunTrigger(target *Workspace, workspaces []*Workspace, organization string) (*RunTrigger, error) {
+	trigger := &RunTrigger{
+		Workspace: target,
+	}
+
+	if rt.SourceID != "" && rt.SourceName != "" {
+		return nil, fmt.Errorf("run trigger source ID and source name cannot both be set")
+	}
+
+	if rt.SourceID != "" {
+		trigger.SourceID = rt.SourceID
+	} else if rt.SourceName != "" {
+		for _, ws := range workspaces {
+			if ws.Name == rt.SourceName {
+				trigger.SourceID = fmt.Sprintf("${tfe_workspace.workspace[%q].id}", ws.Workspace)
+			}
+		}
+
+		if trigger.SourceID == "" {
+			trigger.DataRef = map[string]tfeprovider.DataWorkspace{
+				rt.SourceName: {
```

Comment:
Both this and L53 would break if this contained a space. While that would be [invalid](https://www.terraform.io/docs/cloud/workspaces/naming.html), the action shouldn't fail with a cryptic Terraform syntax error. The guaranteed fix here is to provide any user-supplied strings to `for_each`.

Reply:
If that's too much trouble you can validate the data up front but that's always a bit tricky to rely on